### PR TITLE
Completed javascript-loops-exercise.

### PR DESCRIPTION
**Pull request by vkahlon**
Screen Shot 2021-12-09 at 5 38 04 PM

### CODE REVIEW

File: javascript-loops-practice/add-suffix-to-all.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/* exported addSuffixToAll */
+function addSuffixToAll(words, suffix) {
+  var newWordArray = [];
+  for (var i = 0; i < words.length; i++) {
+    var newWords = '';
+    newWords = words[i] + suffix;
```
Comment (codingcodymiller) [inline javascript-loops-practice/add-suffix-to-all.js:6]: You can do all of this on one line, there's no need for the starting value of an empty string.

File: javascript-loops-practice/filter-out-nulls.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/* exported filterOutNulls */
+function filterOutNulls(values) {
+  var cleanArray = [];
+  var dirtyArray = [];
```
Comment (codingcodymiller) [inline javascript-loops-practice/filter-out-nulls.js:4]: You don't ever do anything with this data, so there's no need to collect it. You only need the `cleanArray`

File: javascript-loops-practice/filter-out-nulls.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  var i = 0;
```
Comment (codingcodymiller) [inline javascript-loops-practice/filter-out-nulls.js:5]: You should just do this in the initialization expression.

File: javascript-loops-practice/filter-out-strings.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+/* exported filterOutStrings */
+/* exported filterOutNulls */
+function filterOutStrings(values) {
+  var stringFreeArray = [];
+  var stringArray = [];
```
Comment (codingcodymiller) [inline javascript-loops-practice/filter-out-strings.js:5]: Same deal, this data is unnecessary to collect.

File: javascript-loops-practice/find-index.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+/* exported findIndex */
+function findIndex(array, value) {
+  var theTruth = 0;
+  for (var i = 0; i < array.length; i++) {
+    if (array[i] === value) {
+      return i;
+    } else if (array[i] !== value) {
+      theTruth = -1;
```
Comment (codingcodymiller) [inline javascript-loops-practice/find-index.js:8]: If you just make the initial value of `theTruth` -1, you can avoid needing this block. Also, funny name, but I think the variable might be better named as something like `notFound` 😉

File: javascript-loops-practice/includes-seven.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+/* exported includesSeven */
+function includesSeven(array) {
+  var doesIncSeven = false;
+  for (var i = 0; i < array.length; i++) {
+    if (array[i] === 7) {
+      return true;
+    } else {
+      doesIncSeven = false;
+    }
```
Comment (codingcodymiller) [inline javascript-loops-practice/includes-seven.js:9]: This block is unnecessary.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

File: javascript-loops-practice/filter-out-nulls.js (hunk 3 above)
Comment (vkahlon) [reply javascript-loops-practice/filter-out-nulls.js:5]: this made me facedesk even harder! even more than the bug I was struggling with

**Comment by vkahlon** [comment]
made the necessary changes. I need to treat !== with more respect next time.

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**